presto-main/src/main/java/com/facebook/presto/cost/CoefficientBasedCostCalculator.java

```diff
@@ -71,16 +72,14 @@ public CoefficientBasedCostCalculator(Metadata metadata)
     }
 
     @Override
-    public Map<PlanNode, PlanNodeCost> calculateCostForPlan(Session session, Map<Symbol, Type> types, PlanNode planNode)
+    public PlanNodeCost calculateCostForNode(Session session, Map<Symbol, Type> types, PlanNode planNode, List<PlanNodeCost> childCosts)
     {
         Visitor visitor = new Visitor(session, types);
-        HashMap<PlanNode, PlanNodeCost> costMap = new HashMap<>();
-        planNode.accept(visitor, costMap);
-        return ImmutableMap.copyOf(costMap);
+        return planNode.accept(visitor, childCosts);
     }
 
     private class Visitor
```

Comment:
Add `visitGroupId` and explicitly throw as an `IllegalArguemntException` there we expect it to be handled by `Memo` before passing `PlanNode` to cost calculator.